### PR TITLE
Check if the indexer service exists when purging search tables

### PR DESCRIPTION
**Pull request by bytehead**
| Q                | A
| -----------------| ---
| Fixed issues     | n/a
| Docs PR or issue | n/a

When the default search indexer is disabled and you click on purge search index you'll get a `ServiceNotFoundException`: `You have requested a non-existent service "contao.search.indexer".`

### CODE REVIEW

**Review by ausi — DISMISSED** [review]

File: core-bundle/src/Resources/contao/library/Contao/Automator.php [hunk 1]
```diff
@@ -35,9 +35,14 @@ public function __construct()
...
+		// The search indexer is disabled
+		if (!System::getContainer()->has('contao.search.indexer'))
+		{
+			return;
+		}
+
 		$searchIndexer = System::getContainer()->get('contao.search.indexer');
 
-		// The search indexer is disabled
 		if (null === $searchIndexer)
```
Comment (Toflar) [inline core-bundle/src/Resources/contao/library/Contao/Automator.php]: This can be removed as well then, no?

**Review by Toflar — CHANGES_REQUESTED** [review]

File: core-bundle/src/Resources/contao/library/Contao/Automator.php (hunk 1 above)
Comment (bytehead) [reply core-bundle/src/Resources/contao/library/Contao/Automator.php]: The `Container::get()` could still return `null`, no?

Comment (Toflar) [reply core-bundle/src/Resources/contao/library/Contao/Automator.php]: If you check for `has()` previously, I don't think that can work, no :)

Comment (bytehead) [reply core-bundle/src/Resources/contao/library/Contao/Automator.php]: Removed in ecc8f31db.

**Review by Toflar — DISMISSED** [review]

**Review by ausi — APPROVED** [review]

**Comment by leofeyer** [comment]
Thank you @bytehead.

**Comment by aschempp** [comment]
Shouldn't we just not offer this option instead in the back end or automator CLI ?

**Comment by bytehead** [comment]
We could indeed. I've recognized that truncating the search tables needs the indexer service. As you have deactivated the search indexer, you're not able to clear the search tables anymore. Wouldn't make it sense to truncate the tables directly in the automator?
https://github.com/contao/contao/blob/f76879682c1280fbc11e6a20497213368e0939ae/core-bundle/src/Resources/contao/library/Contao/Automator.php#L47

**Comment by bytehead** [comment]
Like here: https://github.com/bytehead/contao/commit/d68a3eb06e401f86e73862af0e00658bb49851fa

**Comment by leofeyer** [comment]
We should open a new issue for that and optionally add the "up for discussion" flag.

**Comment by bytehead** [comment]
I'll do :)

**Comment by bytehead** [comment]
See #3326

**Pull request merged**